frameworks/Java/jooby2/src/main/java/com/techempower/App.java

```diff
@@ -1,131 +1,160 @@
-                .prepareStatement(SELECT_WORLD, TYPE_FORWARD_ONLY,
-                    CONCUR_READ_ONLY)) {
-              result.add(findOne(query, id));
+          for (int i = 0; i < result.length; i++) {
+            int id = nextRandom(rnd);
+            try (final PreparedStatement statement = conn.prepareStatement(SELECT_WORLD)) {
+              statement.setInt(1, id);
+              try (ResultSet rs = statement.executeQuery()) {
+                rs.next();
+                result[i] = new World(rs.getInt("id"), rs.getInt("randomNumber"));
+              }
             }
           }
         }
-        ctx.setContentType(json);
-        return ctx.sendBytes(mapper.writeValueAsBytes(result));
+        return ctx
+            .setContentType(JSON)
+            .sendBytes(mapper.writeValueAsBytes(result));
       });
 
       /** Updates: */
       get("/updates", ctx -> {
-        // bound count
-        int count = queries(ctx);
-        List<World> result = new ArrayList<>();
+        World[] result = new World[queries(ctx)];
         Random rnd = ThreadLocalRandom.current();
-        try (Connection conn = ds.getConnection()) {
-          for (int i = 0; i < count; i++) {
-            int id = rndId(rnd);
-            int newRandomNumber = rndId(rnd);
-            try (final PreparedStatement query = conn
-                .prepareStatement(SELECT_WORLD, TYPE_FORWARD_ONLY, CONCUR_READ_ONLY);
-                final PreparedStatement update = conn.prepareStatement(UPDATE_WORLD)) {
-              // find
-              World world = findOne(query, id);
-
-              // update
-              update.setInt(1, newRandomNumber);
-              update.setInt(2, id);
-              update.execute();
-
-              result.add(new World(world.id, newRandomNumber));
+
+        StringJoiner updateSql = new StringJoiner(
+            ", ",
+            "UPDATE world SET randomNumber = temp.randomNumber FROM (VALUES ",
+            " ORDER BY 1) AS temp(id, randomNumber) WHERE temp.id = world.id");
+
+        try (Connection connection = ds.getConnection()) {
+          try (PreparedStatement statement = connection.prepareStatement(SELECT_WORLD)) {
+            for (int i = 0; i < result.length; i++) {
+              statement.setInt(1, nextRandom(rnd));
+              try (ResultSet rs = statement.executeQuery()) {
+                rs.next();
+                result[i] = new World(rs.getInt("id"), rs.getInt("randomNumber"));
+              }
+              // prepare update query
+              updateSql.add("(?, ?)");
             }
           }
+
+          try (PreparedStatement statement = connection.prepareStatement(updateSql.toString())) {
+            int i = 0;
+            for (World world : result) {
+              world.randomNumber = nextRandom(rnd);
+              statement.setInt(++i, world.id);
+              statement.setInt(++i, world.randomNumber);
+            }
+            statement.executeUpdate();
+          }
```

Comment:
This implementation is batching all the updates into a single query, which is against the rules.